be/src/exec/olap_common.h

```diff
@@ -258,6 +258,9 @@ class OlapScanKeys {
     template <class T>
     Status extend_scan_key(ColumnValueRange<T>& range, int32_t max_scan_key_num);
 
+    template <class T>
+    Status extend_scan_key(ColumnValueRange<T>& range, int32_t max_scan_key_num, bool* exact_value);
+
```

Comment:
The implementations of `Status extend_scan_key(ColumnValueRange<T>& range, int32_t max_scan_key_num);` and `Status extend_scan_key(ColumnValueRange<T>& range, int32_t max_scan_key_num, bool* exact_value);` are very similar, I think they can be combined into one, just give `exact_value` a default value `nullptr` when the function is declared. Before assigning a value to `*exact_value`, first determine whether `exact_value` is `nullptr`.

Reply:
To be honest, I considered this way before. Finally I made this because I don't want to mix logics with row-based engine anymore. However, I'm not sure if it's necessary for now and let us wait for more opnions from others :)